src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Fix-GeneratedCode.ps1

```diff
@@ -6,6 +6,12 @@ function Replace-InFile($filePath, $oldText, $newText)
 
 $generatedFolder = ".\GeneratedSearchIndex"
 
+# Delete any extra files generated for types that are shared between SearchServiceClient and SearchIndexClient.
```

Comment:
They are separate because we actually want two different clients. Our programming model works like this:

    var index = new Index() { Name = "test", ... };
    var serviceClient = new SearchServiceClient(...);
    serviceClient.Indexes.Create(index);
    ISearchIndexClient indexClient = serviceClient.Indexes.GetClient("test");
    var results = indexClient.Documents.Search(...);

Every index has its own schema, so we have overloads of `Search` and friends that take a generic type parameter for a customer-supplied model class, and other overloads that just return property bags (`Dictionary<string, object>` more or less).

Eventually we hope that `SearchServiceClient` can be 100% generated code, but `SearchIndexClient` will never be.